### PR TITLE
Py35 support (close #110)

### LINKED ISSUE #110: Support Python 3.5
Do or do not?

Thinkable issue

### 1. Deploy

Our (old) deploy logic rely on python version checking.

### 2. New feature

I think there is no problem because we didn't use `coroutine` or matrix calc with numpy.

Just, We must consider [typing module](https://docs.python.org/3.5/library/typing.html).

### PR DESCRIPTION
**Pull request by item4**
**DO NOT MERGE IT NOW!**

It must rebase after #111

### CODE REVIEW

File: hooks/pre-commit [hunk 1]
```diff
@@ -6,7 +6,7 @@ else
     VERSION=$(python3 -c 'import sys; v = sys.version_info; print("{}.{}".format(v[0], v[1]));')
 fi
 python3 -m flake8 -j auto
-if [ $VERSION = '3.4' ]; then
+if [ $VERSION != '3.3' ]; then
```
Comment (miaekim) [inline hooks/pre-commit]: why don't you use `$VERSION = '3.4' or $VESRION = '3.5'`? I am afraid lower version of python might pass this `if`.

Comment (item4) [inline hooks/pre-commit]: @miaekim If py36, 37, 38.... will release, our code must be `$VERSION = '3.4' or $VESRION = '3.5' or $VESRION = '3.6' or $VESRION = '3.7' or $VESRION = '3.8' or ...`.

Comment (miaekim) [inline hooks/pre-commit]: Then you could write
`float($VERSION) > 3.3`

Comment (item4) [inline hooks/pre-commit]: It's bash script, not Python.

**Pull request closed without merging**